data/visit_pytar.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Lawrence Livermore National Security, LLC and other VisIt
+# Project developers.  See the top-level LICENSE file for dates and other
+# details.  No copyright assignment is required to contribute to VisIt.
+
+
+# https://martinheinz.dev/blog/57
+
+import os, sys, tarfile
+
+#
+# Output a usage message with optional error messag
+#
+def usage(errmsg=""):
+    print(\
+"""
+%s
+Usage: Basic tar-compatible archiver for VisIt test data archives
+    using Python3 tarfile & lzma modules with (xz) compression.
+
+    Create: visit_pytar c|C <batch>_test_data.tar.xz
+    List:   visit_pytar t   <batch>_test_data.tar.xz
+    Expand: visit_pytar x|X <batch>_test_data.tar.xz
+
+    Where <batch> is an arbitrary string naming a batch of related
+    data files. Usually, it is used to indicate the underlying data
+    format all the files use.
+
+    For Create, (c) option, a matching directory name of the form
+    <batch>_test_data must already exist. Lowercase c Create fails
+    if <batc>_test_data.tar.xz already exists. Uppercase C ignores
+    this check.
+
+    For Expand, Lowercase x fails if <batch>_test_data already
+    exists. Uppercase X ignores this check.
+
+    This tooling is provided only as a convenience. More advanced
+    archiving and compression operations are likely possible using
+    your platform's native tooling. For example, on many systems
+    the command...
+
+    tar cvf - my_test_data | xz -9e -T0 - > my_test_data.tar.xz
+
+    will use tar and xv commands in a pipe with maximal compression
+    and as many threads as actual hardware cores.
+"""%("***Error***: %s\n"%errmsg if errmsg else "")
+)
+    sys.exit(1)
+
+if len(sys.argv) == 1:
+    usage()
+elif sys.version_info.major != 3:
+    usage('Python version 3 is required')
+
+elif 'c' in sys.argv[1].lower(): # Create
+    if not sys.argv[2].endswith('_test_data.tar.xz'):
+        usage('"%s" must be of the form <dbformat>_test_data.tar.xz'%sys.argv[2])
+    if 'c' in sys.argv[1] and os.path.exists(sys.argv[2]):
+        usage('%s must not already exist'%sys.argv[2])
+    if not os.path.isdir(sys.argv[2][:-7]): 
+        usage('"%s" must be a pre-existing directory'%sys.argv[2][:-7])
+    tf = tarfile.open(sys.argv[2], 'w:xz', preset=9)
+    tf.add(sys.argv[2][:-7])
+    tf.close()
+
+elif 't' in sys.argv[1]: # List
+    if not os.path.exists(sys.argv[2]):
+        usage('%s does not exist'%sys.argv[2])
+    tf = tarfile.open(sys.argv[2], 'r')
+    tf.list()
+    tf.close()
+
+elif 'x' in sys.argv[1].lower(): # Expand
+    if not os.path.exists(sys.argv[2]):
+        usage('%s does not exist'%sys.argv[2])
+    if 'x' in sys.argv[1] and os.path.isdir(sys.argv[2][:-7]):
```

Comment:
This script needs to account for out-of-source builds.  This particular test looks to see if the tarball has already been extracted into the same location as the tarball itself.
I was trying to test on Windows, and had some tarballs extracted in `visit/data`, but I was attempting to **build** `testdata` which is extracting to  `Build/testdata` and the script threw an error and stopped processing.

Reply:
What do you think of either passing in as an argument to the script the extraction location, or use 'cwd'.
That way test 'already exists' test would be testing against the correct location, which isn't necessarily the same directory as the tarball.

Reply:
Yeah, that makes sense I think. I think it would be best if `visit_pytar.py` behaved more or less just like `tar`, at least in terms of how it interprets args (but perhaps with only a limited set of args it knows...I don't wanna promise to code the python equiv of the `tar` command).